livedata/src/main/java/io/getstream/chat/android/livedata/service/SyncService.kt

```diff
@@ -0,0 +1,64 @@
+package io.getstream.chat.android.livedata.service
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.Service
+import android.content.Intent
+import android.os.Build
+import android.os.IBinder
+import androidx.core.app.NotificationCompat
+import io.getstream.chat.android.client.events.NewMessageEvent
+import io.getstream.chat.android.livedata.ChatDomain
+import io.getstream.chat.android.livedata.ChatNotificationConfigOffline.Companion.EXTRA_CID
+
+class SyncService: Service() {
+
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        createNotificationChannel()
+
+        NotificationCompat.Builder(this, CHANNEL_ID)
+            .setAutoCancel(true)
+            .build()
+            .apply {
+                startForeground(NOTIFICATION_ID, this)
+            }
+
+        try {
+            intent?.getStringExtra(EXTRA_CID)?.let {
+                // TODO: ChatDomain is null when app gets killed.
+                // That's why it's wrapped in try clause atm.
+                // Need to store api key to re-create ChatClient + ChatDomain here.
+                ChatDomain.instance().apply {
```

Comment:
All this logic (For now very simple, but it will do some extra job) can be extracted into another collaborator, then we are able to test our logic without needing to instantiate our service

Reply:
Good idea, I'll extract it to a new use case perhaps, in next PR